src/chunkserver/chunkserver-common/default_disk_manager.cc

```diff
@@ -0,0 +1,305 @@
+	}
+
+	IDisk *currentDisk = DiskNotFound;
+
+	if (configuration.isZoned) {
+		currentDisk = pluginManager.createDisk(configuration);
+	} else {
+		currentDisk = new CmrDisk(configuration);
+	}
+
+	if (currentDisk == DiskNotFound) {
+		throw InitializeException("HDD configuration line not valid: " +
+		                          hddCfgLine);
+	}
+
+	{
+		// Ensure meta + data path remain consistent between reloads
+		std::lock_guard disksLockGuard(gDisksMutex);
+
+		for (const auto &disk : gDisks) {
+			if (disk->metaPath() == currentDisk->metaPath()) {
+				if (disk->dataPath() != currentDisk->dataPath()) {
+					throw InitializeException(
+					    "Combination of metadata and "
+					    "data paths have changed between "
+					    "reloads for line: " +
+					    hddCfgLine);
+				}
+			}
+		}
+
+		// Lock is not needed if the disk already exists in memory (reload)
+		lockNeeded = 1;
+		for (const auto &disk : gDisks) {
+			if (disk->metaPath() == currentDisk->metaPath()) {
+				lockNeeded = 0;
+				break;
+			}
+		}
+	}
+
+	try {
+		currentDisk->createPathsAndSubfolders();
+		{
+			std::lock_guard disksLockGuard(gDisksMutex);
+			currentDisk->createLockFiles(lockNeeded, gDisks);
+		}
+	} catch (const InitializeException &ie) {
+		delete currentDisk;
+		throw InitializeException(ie.what());
+	}
+
+	std::unique_lock disksUniqueLock(gDisksMutex);
+
+	// This loop is used at reload time, we need to update the already existing
+	// Disks' properties according to the (probably new) hdd.cfg file.
+	for (auto &disk : gDisks) {
+		if (disk->metaPath() ==
+		    currentDisk->metaPath()) {  // Disk already in memory
+			if (disk->isDamaged()) {
+				disk->setScanState(IDisk::ScanState::kNeeded);
+				disk->setScanProgress(0);
+				disk->setIsDamaged(currentDisk->isDamaged());
+				disk->setAvailableSpace(0ULL);
+				disk->setTotalSpace(0ULL);
+				disk->setLeaveFreeSpace(disk::gLeaveFree);
+				disk->getCurrentStats().clear();
+				for (int l = 0; l < disk::kStatsHistoryIn24Hours; ++l) {
+					disk->stats()[l].clear();
+				}
+				disk->setStatsPos(0);
+				for (int l = 0; l < disk::kLastErrorSize; ++l) {
+					disk->lastErrorTab()[l].chunkid = 0ULL;
+					disk->lastErrorTab()[l].timestamp = 0;
+				}
+				disk->setLastErrorIndex(0);
+				disk->setLastRefresh(0);
+				disk->setNeedRefresh(true);
+			} else {
+				if (disk->isMarkedForRemoval() !=
+				        currentDisk->isMarkedForRemoval() ||
+				    disk->isReadOnly() != currentDisk->isReadOnly()) {
+					// important change - chunks need to be send to master again
+					disk->setScanState(IDisk::ScanState::kSendNeeded);
+				}
+			}
+
+			disk->setWasRemovedFromConfig(false);
+			disk->setIsReadOnly(currentDisk->isReadOnly());
+			disk->setIsMarkedForRemoval(currentDisk->isMarkedForRemoval());
+			disksUniqueLock.unlock();
+
+			delete currentDisk;  // Also deletes the lock files (smart pointer)
+			return 1;
+		}
+	}
+
+	gDisks.emplace_back(currentDisk);
+	gResetTester = true;
+
+	return 2;
```

Comment:
Consider to return constant names instead of values. Or add a little comment about their meaning or possible values returned by the function if you consider it necessary

Reply:
There is no change in the logic in this PR. I agree with the comment, but this function will change soon. I did not change the original content.